lib/presentation/widget/destination_picker/destination_picker_widget.dart

```diff
@@ -0,0 +1,274 @@
+                  )),
+            ],
+          ),
+        ));
+  }
+
+  Widget _buildSharedSpacesList(
+      BuildContext context, DestinationPickerState state) {
+    return state.viewState.fold(
+        (failure) => RefreshIndicator(
+            onRefresh: () async =>
+                _destinationPickerViewModel.getAllSharedSpaces(),
+            child: failure is SharedSpaceFailure
+                ? BackgroundWidgetBuilder()
+                    .key(Key('shared_space_error_background'))
+                    .image(SvgPicture.asset(_imagePath.icUnexpectedError,
+                        width: 120, height: 120, fit: BoxFit.fill))
+                    .text(AppLocalizations.of(context)
+                        .common_error_occured_message)
+                    .build()
+                : _buildSharedSpacesListView(context, state.sharedSpacesList)),
+        (success) => success is LoadingState
+            ? _buildSharedSpacesListView(context, state.sharedSpacesList)
+            : RefreshIndicator(
+                onRefresh: () async =>
+                    _destinationPickerViewModel.getAllSharedSpaces(),
+                child: _buildSharedSpacesListView(
+                    context, state.sharedSpacesList)));
+  }
+
+  Widget _buildSharedSpacesListView(
+      BuildContext context, List<SharedSpaceNodeNested> sharedSpacesList) {
+    if (sharedSpacesList.isEmpty) {
+      return _buildNoWorkgroupYet(context);
+    } else {
+      return ListView.builder(
+        key: Key('shared_spaces_list'),
+        padding: EdgeInsets.zero,
+        itemCount: sharedSpacesList.length,
+        itemBuilder: (context, index) {
+          return _buildSharedSpaceListItem(context, sharedSpacesList[index]);
+        },
+      );
+    }
+  }
+
+  Widget _buildNoWorkgroupYet(BuildContext context) {
+    return BackgroundWidgetBuilder()
+        .key(Key('shared_space_no_workgroup_yet'))
+        .image(SvgPicture.asset(_imagePath.icSharedSpaceNoWorkGroup,
+            width: 120, height: 120, fit: BoxFit.fill))
+        .text(AppLocalizations.of(context).do_not_have_any_workgroup)
+        .build();
+  }
+
+  Widget _buildSharedSpaceListItem(
+      BuildContext context, SharedSpaceNodeNested sharedSpace) {
+    return ListTile(
+      leading: Column(mainAxisAlignment: MainAxisAlignment.center, children: [
+        SvgPicture.asset(_imagePath.icSharedSpace,
+            width: 20, height: 24, fit: BoxFit.fill)
+      ]),
+      title: Transform(
+        transform: Matrix4.translationValues(-16, 0.0, 0.0),
+        child: _buildSharedSpaceName(sharedSpace.name),
+      ),
+      onTap: () {
+        _destinationPickerViewModel.openSharedSpaceInside(sharedSpace);
+      },
+    );
+  }
+
+  Widget _buildSharedSpaceName(String sharedSpaceName) {
+    return Text(
+      sharedSpaceName,
+      maxLines: 1,
+      style: TextStyle(fontSize: 14, color: AppColor.documentNameItemTextColor),
+    );
+  }
+}
```

Comment:
we can not reuse any of this from SharedSpace?

Reply:
Then we need to create some widget builders to reuse this, but I don't think this is necessary in this case :(